pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -1026,41 +1066,101 @@ func registriesConfigIgnition(templateDir string, controllerConfig *mcfgv1.Contr
 	return &registriesIgn, nil
 }
 
-// getValidScopePolicies returns a map[scope]policyRequirement from ClusterImagePolicy
-func getValidScopePolicies(clusterImagePolicies []*apicfgv1alpha1.ClusterImagePolicy) (map[string]signature.PolicyRequirements, error) {
+// getValidScopePolicies returns a map[scope]policyRequirement from ClusterImagePolicy, a map[scope][namespace]policyRequirement from ImagePolicy CRs.
+// It skips ImagePolicy scopes that conflict with ClusterImagePolicy scopes and logs the conflicting scopes in the ImagePolicy Status.
+func getValidScopePolicies(clusterImagePolicies []*apicfgv1alpha1.ClusterImagePolicy, imagePolicies []*apicfgv1alpha1.ImagePolicy, ctrl *Controller) (map[string]signature.PolicyRequirements, map[string]map[string]signature.PolicyRequirements, error) {
 	clusterScopePolicies := make(map[string]signature.PolicyRequirements)
+	namespacePolicies := make(map[string]map[string]signature.PolicyRequirements)
+
 	for _, clusterImagePolicy := range clusterImagePolicies {
 		sigstoreSignedPolicyItem, err := policyItemFromSpec(clusterImagePolicy.Spec.Policy)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		for _, scope := range clusterImagePolicy.Spec.Scopes {
 			scopeStr := string(scope)
 			clusterScopePolicies[scopeStr] = append(clusterScopePolicies[scopeStr], sigstoreSignedPolicyItem)
 		}
 	}
 
-	return clusterScopePolicies, nil
+	for _, imagePolicy := range imagePolicies {
+		namespace := imagePolicy.ObjectMeta.Namespace
+		sigstoreSignedPolicyItem, err := policyItemFromSpec(imagePolicy.Spec.Policy)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		var conflictScopes []string
+		for _, scope := range imagePolicy.Spec.Scopes {
+			scopeStr := string(scope)
+			nestedInClusterScope := false
+			for clusterScope := range clusterScopePolicies {
+				if runtimeutils.ScopeIsNestedInsideScope(scopeStr, clusterScope) {
+					conflictScopes = append(conflictScopes, scopeStr)
+					nestedInClusterScope = true
```

Comment:
nestedInClusterScope can just continue here, right? Since we have line 1104?

Reply:
Do you suggest this?
```go
					
if runtimeutils.ScopeIsNestedInsideScope(scopeStr, clusterScope) {
					conflictScopes = append(conflictScopes, scopeStr)
					nestedInClusterScope = true
					continue
				}
			}
```
I think we can break here to avoid add scopeSte to the list twice.

Reply:
Ah I see... We want to continue on the outer scope.

on line 1097, could add a outerScope: label... and then instead of breaking, continue outerScope... That would delete lines 1008-1110, and make the intent perhaps more clear.

Reply:
```go

		var conflictScopes []string
    outerScope:
		for _, scope := range imagePolicy.Spec.Scopes {
			scopeStr := string(scope)			
			for clusterScope := range clusterScopePolicies {
				if runtimeutils.ScopeIsNestedInsideScope(scopeStr, clusterScope) {
					conflictScopes = append(conflictScopes, scopeStr)					
					continue outerScope
				}
			}
			if _, ok := namespacePolicies[scopeStr]; !ok {
				namespacePolicies[scopeStr] = make(map[string]signature.PolicyRequirements)
			}
			namespacePolicies[scopeStr][namespace] = append(namespacePolicies[scopeStr][namespace], sigstoreSignedPolicyItem)
		}
```